nautobot/extras/models/metadata.py

```diff
@@ -135,3 +141,298 @@ def save(self, *args, **kwargs):
+        on_delete=models.PROTECT,
+        related_name="object_metadatas",
+        blank=True,
+        null=True,
+    )
+    scoped_fields = JSONArrayField(
+        base_field=models.CharField(
+            max_length=CHARFIELD_MAX_LENGTH,
+        ),
+        help_text="List of scoped fields, only direct fields on the model",
+    )
+    _value = models.JSONField(
+        blank=True,
+        null=True,
+        help_text="Relevant data value to an object field or a set of object fields",
+    )
+    assigned_object_type = models.ForeignKey(to=ContentType, on_delete=models.SET_NULL, null=True, related_name="+")
+    assigned_object_id = models.UUIDField(db_index=True)
+    assigned_object = GenericForeignKey(ct_field="assigned_object_type", fk_field="assigned_object_id")
+
+    objects = ObjectMetadataManager()
+    natural_key_field_names = ["pk"]
+    documentation_static_path = "docs/user-guide/platform-functionality/metadata.html"
+
+    class Meta:
+        ordering = ["metadata_type"]
+
+        indexes = [
+            models.Index(
+                name="assigned_object",
+                fields=["assigned_object_type", "assigned_object_id"],
+            ),
+            models.Index(
+                name="assigned_object_contact",
+                fields=["assigned_object_type", "assigned_object_id", "contact"],
+            ),
+            models.Index(
+                name="assigned_object_team",
+                fields=["assigned_object_type", "assigned_object_id", "team"],
+            ),
+        ]
+
+    @property
+    def value(self):
+        if self.metadata_type.data_type == MetadataTypeDataTypeChoices.TYPE_CONTACT_TEAM:
+            return self.contact or self.team
+        else:
+            return self._value
+
+    @value.setter
+    def value(self, v):
+        if self.metadata_type.data_type == MetadataTypeDataTypeChoices.TYPE_CONTACT_TEAM:
+            if isinstance(v, Contact):
+                self.contact = v
+                self.team = None
+            elif isinstance(v, Team):
+                self.team = v
+                self.contact = None
+            else:
+                raise ValidationError(
+                    f"{v} is an invalid value for metadata type data type {self.metadata_type.data_type}"
+                )
+        else:
+            self._value = v
+        self.clean()
+
+    def __str__(self):
+        return f"{self.metadata_type} - {self.assigned_object} - {self.value}"
+
+    def clean(self):
+        """
+        Validate a value according to the field's type validation rules.
+
+        Returns the value, possibly cleaned up
+        """
+        super().clean()
+        value = self._value
+        metadata_type_data_type = self.metadata_type.data_type
+        if (
+            isinstance(value, list)
+            and len(value) == 1
+            and self.metadata_type.data_type != MetadataTypeDataTypeChoices.TYPE_MULTISELECT
+        ):
+            value = value[0]
```

Comment:
This feels like a kludge to me. Is it really needed?

Reply:
Yeah, I don't like the pattern here either. But this is for the csv create case. `test_recreate_object_csv` caught this edge case. Since the value field is a `JSONField` on the serializer, when you pass a `float`/`str`, etc, the value field will cast them into a list. Declaring it as a `CharField` solves it for the `float/str` case but not the other types. I wonder if a custom `SerializerMethodField` will solve it.
```
['1460.352280751']
float
F
======================================================================
FAIL: test_recreate_object_csv (nautobot.extras.tests.test_api.ObjectMetadataTest)
CSV export an object, delete it, and recreate it via CSV import.
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/source/nautobot/core/testing/api.py", line 707, in test_recreate_object_csv
    self.assertHttpStatus(response, status.HTTP_201_CREATED, csv_data)
  File "/source/nautobot/core/testing/mixins.py", line 179, in assertHttpStatus
    self.assertIn(response.status_code, expected_status, err_message)
AssertionError: 500 not found in [201] : display,id,object_type,natural_slug,assigned_object_type,assigned_object,value,scoped_fields,assigned_object_id,metadata_type__name,contact__name,contact__phone,contact__email,team__name,team__phone,team__email,created,last_updated
Accounting technician - Advertising account executive (rachellopez@example.net) - 1460.352280751,86e74add-c680-4d83-86d9-2e684cfc6680,extras.objectmetadata,86e74add-c680-4d83-86d9-2e684cfc6680_86e7,extras.team,25b3f8b0-9fa7-483f-af5b-e3f42e4f0a9a,1460.352280751,"zVLNIBubsHkSgJYJMIxV,dqtBPZXVBeDdQtsIaAGY,bvzwmonNhPOjFdxvRlca,LZQgYNkkTrSifkLZcxxD,uqIdNrochpnVSMgTRVyS,kWvhuuqTsARuOLHJUZkS,tjtdMZFXpCQcjlpFUWeF",25b3f8b0-9fa7-483f-af5b-e3f42e4f0a9a,Accounting technician,NoObject,NoObject,NoObject,NoObject,NoObject,NoObject,2024-06-27T20:39:00.875000Z,2024-06-27T20:39:00.875000Z

Expected HTTP status(es) [201]; received 500:
{"error": "float() argument must be a string or a real number, not 'list'", "exception": "TypeError", "nautobot_version": "2.3.0a1", "python_version": "3.10.14"}
```

Reply:
Thanks for the explanation. I think maybe just adding a comment to this effect may be fine for the moment, and maybe opening a followup issue to investigate whether something needs to be tweaked on the serializer.

Reply:
https://github.com/nautobot/nautobot/issues/5866